networkit/cpp/centrality/GroupClosenessGrowShrinkImpl.cpp

```diff
@@ -16,12 +16,10 @@
 namespace NetworKit {
 namespace GroupClosenessGrowShrinkDetails {
 template <class WeightType>
-GroupClosenessGrowShrinkImpl<WeightType>::GroupClosenessGrowShrinkImpl(const Graph &G,
-                                                                       std::vector<node> initGroup,
-                                                                       bool extended,
-                                                                       count insertions,
-                                                                       count maxIterations)
-    : G(&G), group(std::move(initGroup)), extended(extended),
+GroupClosenessGrowShrinkImpl<WeightType>::GroupClosenessGrowShrinkImpl(
+    const Graph &G, std::vector<node> &&initGroup, bool extended, count insertions,
```

Comment:
Changing parameter `initGroup` to r-value reference does look odd. If a l-value is given, does this still work? In the unittests only [iterator-based constructor](https://github.com/networkit/networkit/blob/master/networkit/cpp/centrality/test/CentralityGTest.cpp#L1937) is used.

Reply:
I think this is ok because it's the private implementation, not the public interface. The algorithms needs to change `group` anyway, so instead of copying a const ref it just takes an r-value ref.

Reply:
Ah, it works because the public interface does make a copy, which is then handed over to the private interface via r-value ref. Makes sense.

Reply:
Let the constructor take the group by value in this case (and move the value to the member variable).